File: src/Lavcode.Asp/Dtos/CreateFolderDto.cs
using System.ComponentModel.DataAnnotations;

namespace Lavcode.Asp.Dtos
{
    public class CreateFolderDto
    {
        [Required]
        public string Name { get; set; } = null!;

        [Required]
        public UpsertIconDto Icon { get; set; } = null!;
    }
}

File: src/lavcode-node/src/actions/folder/dtos/create-folder.dto.ts
import { V } from "@halsp/validator";
import { UpsertIconDto } from "../../../dtos/upsert-icon.dto";

export class CreateFolderDto {
  @V().Required().IsString()
  name!: string;

  @V().Required()
  icon!: UpsertIconDto;
}
